src/commands/ScriptCommandRelay.cpp

```diff
@@ -57,6 +58,8 @@ static int ExecCommand(wxString *pIn, wxString *pOut, bool fromMain)
       // Wait for and retrieve the response
       *pOut = builder.GetResponse();
    }
+   else
+      *pOut = wxString{};
```

Comment:
This seems like a change of behavior that needs to be tested accurately.

Reply:
I have not tested it but this only involves mod-script-pipe which I believe could crash without this change in case of macOS with the application still running but not project window open.

The unchecked assumption that GetActiveProject returns non-null is not a safe assumption and I recall removing it in other case.

Reply:
A review of the uses of GetActiveProject() isn't strictly in the scope of the problem of this library extraction.  Still I did it incidentally because I recall other bugs with use of that functon, and I changed the return type to a weak pointer to be sure there not not dangling non-null pointers.

I have really disliked GetActiveProject() and in the past eliminated many uses of it in favor instead of a project reference passed in CommandContext.

Reply:
Still GetActiveProject() will require use in a few places where CommandContext is itself constructed.